snippets/elixir.snippets

```diff
@@ -265,22 +265,27 @@ snippet genserver basic genserver structure
 	use GenServer
 
 	@doc false
-	def start_link do
-		GenServer.start_link(__MODULE__, ${1:Map.new})
+	def start_link(init_args) do
+		GenServer.start_link(__MODULE__, init_args, name: __MODULE__)
 	end
 
 	@impl true
-	def init(state) do
-		{:ok, state}
+	def init(_init_args) do
+		{:ok, []}
```

Comment:
Let's keep with the previous pattern of passing in the state.

Reply:
The reason I changed it is that although it might be state you're passing it, it might not be - generate it is some sort of initialisation info, which might include initial state.

Reply:
Yes please

Reply:
Done :)